content/en/blog/_posts/2021-11-18-persistentvolume-reclaim-policy-improvements.md

```diff
@@ -0,0 +1,200 @@
+
+
+## How did reclaim work in previous Kubernetes releases?
+
+[PersistentVolumeClaim](/docs/concepts/storage/persistent-volumes/#Introduction) (or PVC for short) is
+a request for storage by a user. A PV and PVC are considered [Bound](/docs/concepts/storage/persistent-volumes/#Binding)
+if there is a newly created PV or a matching PV is found. The PVs themselves are
+backed by a volume allocated by the storage backend. 
+
+Normally, if the volume is to be deleted, then the expectation is to delete the
+PVC for a bound PV-PVC pair. However, there are no restrictions to delete a PV
+prior to deleting a PVC.
+
+First, I'll demonstrate the behavior for clusters that are running an older version of Kubernetes.
+
+#### Create PVC and wait for it to get bound
+
+Create the PVC defined by a `pvc.yaml`
+```
+kubectl create -f pvc.yaml 
+```
+The following output is seen when the PVC is created:
+```
+persistentvolumeclaim/example-vanilla-block-pvc created
+```
+Retrieve the PVC
+```
+kubectl get pvc example-vanilla-block-pvc
+```
+The following output shows the PVC and it's `Bound` PV, the PV is shown under the `VOLUME` column:
+```
+NAME                        STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS               AGE
+example-vanilla-block-pvc   Bound    pvc-6791fdd4-5fad-438e-a7fb-16410363e3da   5Gi        RWO            example-vanilla-block-sc   19s
+```
+
+#### Delete PV
+
+When I try to delete a bound PV, the cluster blocks and the `kubectl` tool does
+not return back control to the shell; for example:
+
+```
+kubectl delete pv pvc-6791fdd4-5fad-438e-a7fb-16410363e3da
+```
+
+```
+persistentvolume "pvc-6791fdd4-5fad-438e-a7fb-16410363e3da" deleted
+^C
+```
+
+Retrieving the PV:
+```
+kubectl get pv pvc-6791fdd4-5fad-438e-a7fb-16410363e3da
+```
+
+It can be observed that the PV is in `Terminating` state
+```
+NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS        CLAIM                               STORAGECLASS               REASON   AGE
+pvc-6791fdd4-5fad-438e-a7fb-16410363e3da   5Gi        RWO            Delete           Terminating   default/example-vanilla-block-pvc   example-vanilla-block-sc            2m23s
+```
+
+#### Delete PVC
+
+```
+kubectl delete pvc example-vanilla-block-pvc
+```
+
+The following output is seen if the PVC gets successfully deleted:
+```
+persistentvolumeclaim "example-vanilla-block-pvc" deleted
+```
+
+The PV object from the cluster also gets deleted. When attempting to retrieve the PV
+it will be observed that the PV is no longer found:
+
+```
+kubectl get pv pvc-6791fdd4-5fad-438e-a7fb-16410363e3da
+```
+
+```
+Error from server (NotFound): persistentvolumes "pvc-6791fdd4-5fad-438e-a7fb-16410363e3da" not found
+```
+
+Although the PV is deleted the underlying storage resource is not deleted, and
+needs to be removed manually.
+
+To sum it up, Reclaim policy associated with the Persistent Volume is currently
```

Comment:
Reclaim policy -> the reclaim policy

Reply:
```suggestion
To sum it up: if you use Kubernetes v1.22 or earlier, a reclaim policy associated
with a PersistentVolume is
```

“currently” is going to be misleading here.